agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/common/TelemetryTruncation.java

```diff
@@ -21,28 +21,60 @@
 
 package com.microsoft.applicationinsights.agent.internal.common;
 
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 public class TelemetryTruncation {
 
-  private static final WarningLogger warningLogger =
-      new WarningLogger(TelemetryTruncation.class, "Telemetry was truncated");
+  private static final Logger logger = LoggerFactory.getLogger(TelemetryTruncation.class);
+
+  private static final Set<String> alreadyLoggedAttributeNames = ConcurrentHashMap.newKeySet();
+  private static final Set<String> alreadyLoggedPropertyKeys = ConcurrentHashMap.newKeySet();
 
   public static String truncateTelemetry(String value, int maxLength, String attributeName) {
     if (value == null || value.length() <= maxLength) {
       return value;
     }
-    warningLogger.recordWarning("truncated " + attributeName + ": " + value);
+    if (!alreadyLoggedAttributeNames.add(attributeName)) {
+      // this can be expected, so don't want to flood the logs with a lot of these
+      // (and don't want to log the full value, e.g. sql text > 8192 characters)
+      logger.warn(
+          "truncated {} attribute value to {} characters (this message will only be logged once"
+              + " per attribute name, and only for at most 10 different attribute names): {}",
+          attributeName,
+          maxLength,
+          trimTo80(value));
+    }
+    logger.debug(
```

Comment:
should this in the else block?  without it, it will get log twice, warn and debug for the very first time.

Reply:
the debug message has more detail (non-trimmed value)

Reply:
in that case, do we need to trim in warn logger level? 
since it's only logging once, i think it's fine to show details.

Reply:
I think logging an (expected) sql text > 8192 characters (like in #1964) could still be annoying enough for users to ask to suppress it (trying to make this warning not be so bad for the expected cases)